File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/mobilegift/ui/MobileGiftListFragment.java
//package com.pplus.prnumberbiz.apps.mobilegift.ui;
//
//
//import android.content.Context;
//import android.content.Intent;
//import android.graphics.Rect;
//import android.os.Bundle;
//import android.support.annotation.DimenRes;
//import android.support.annotation.NonNull;
//import android.support.v7.widget.GridLayoutManager;
//import android.support.v7.widget.RecyclerView;
//import android.view.View;
//import android.widget.TextView;
//
//import com.pplus.prnumberbiz.Const;
//import com.pplus.prnumberbiz.R;
//import com.pplus.prnumberbiz.apps.common.builder.AlertBuilder;
//import com.pplus.prnumberbiz.apps.common.builder.OnAlertResultListener;
//import com.pplus.prnumberbiz.apps.common.ui.base.BaseFragment;
//import com.pplus.prnumberbiz.apps.mobilegift.data.MobileGiftAdapter;
//import com.pplus.prnumberbiz.core.code.common.EnumData;
//import com.pplus.prnumberbiz.core.network.ApiBuilder;
//import com.pplus.prnumberbiz.core.network.model.dto.MobileGift;
//import com.pplus.prnumberbiz.core.network.model.dto.MobileGiftCategory;
//import com.pplus.prnumberbiz.core.network.model.response.NewResultResponse;
//
//import java.util.HashMap;
//import java.util.Map;
//
//import network.common.PplusCallback;
//import retrofit2.Call;
//
//public class MobileGiftListFragment extends BaseFragment<MobileGiftActivity>{
//
//
//    public MobileGiftListFragment(){
//        // Required empty public constructor
//    }
//
//    public static MobileGiftListFragment newInstance(MobileGiftCategory category){
//
//        MobileGiftListFragment fragment = new MobileGiftListFragment();
//        Bundle args = new Bundle();
//        args.putParcelable(Const.CATEGORY, category);
//        fragment.setArguments(args);
//        return fragment;
//    }
//
//    @Override
//    public String getPID(){
//
//        return null;
//    }
//
//    @Override
//    public int getLayoutResourceId(){
//
//        return R.layout.fragment_mobile_gift_list;
//    }
//
//    private MobileGiftAdapter mAdapter;
//    GridLayoutManager mLayoutManager;
//
//    private int mPage = 1, mTotalCount = 0;
//    private boolean mLockListView = true;
//
//    private TextView text_total_count, text_sort;
//    private MobileGiftCategory mCategory;
//    private EnumData.MobileGiftSortType mSort;
//
//    @Override
//    public void onCreate(Bundle savedInstanceState){
//
//        super.onCreate(savedInstanceState);
//        if(getArguments() != null) {
//            mCategory = getArguments().getParcelable(Const.CATEGORY);
//        }
//    }
//
//    @Override
//    public void initializeView(View container){
//
//        RecyclerView recyclerView = (RecyclerView) container.findViewById(R.id.recycler_mobile_gift_list);
//        mLayoutManager = new GridLayoutManager(getActivity(), 2);
//        recyclerView.setLayoutManager(mLayoutManager);
//        recyclerView.addItemDecoration(new ItemOffsetDecoration(getActivity(), R.dimen.height_16));
//        mAdapter = new MobileGiftAdapter(getActivity());
//        recyclerView.setAdapter(mAdapter);
//
//        text_total_count = (TextView) container.findViewById(R.id.text_mobile_gift_total_count);
//        text_total_count.setText(getString(R.string.format_total_with_count, 0));
//
//        text_sort = (TextView)container.findViewById(R.id.text_mobile_gift_sort);
//        text_sort.setOnClickListener(this);
//
//        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener(){
//
//            int pastVisibleItems, visibleItemCount, totalItemCount;
//
//            @Override
//            public void onScrolled(RecyclerView recyclerView, int dx, int dy){
//
//                super.onScrolled(recyclerView, dx, dy);
//                visibleItemCount = mLayoutManager.getChildCount();
//                totalItemCount = mLayoutManager.getItemCount();
//                pastVisibleItems = mLayoutManager.findFirstVisibleItemPosition();
//                if(!mLockListView) {
//                    if((totalItemCount < mTotalCount) && (visibleItemCount + pastVisibleItems) >= (totalItemCount * Const.NEXT_RECYCLERVIEW_MARGIN)) {
//                        mPage++;
//                        listCall(mPage);
//                    }
//                }
//            }
//        });
//
//        mAdapter.setOnItemClickListener(new MobileGiftAdapter.OnItemClickListener(){
//
//            @Override
//            public void onItemClick(int position){
//
//                Intent intent = new Intent(getActivity(), MobileGiftDetailActivity.class);
//                intent.putExtra(Const.DATA, mAdapter.getItem(position));
//                intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
//                startActivity(intent);
//            }
//        });
//
//        mSort = EnumData.MobileGiftSortType.recent;
//        getCount();
//    }
//
//    @Override
//    public void init(){
//
//    }
//
//    @Override
//    public void onClick(View v){
//
//        super.onClick(v);
//        switch (v.getId()){
//            case R.id.text_mobile_gift_sort:
//                AlertBuilder.Builder builder = new AlertBuilder.Builder();
//                builder.setContents(getString(R.string.word_sort_recent), getString(R.string.word_sort_high_price), getString(R.string.word_sort_low_price));
//                builder.setStyle_alert(AlertBuilder.STYLE_ALERT.LIST_CENTER).setOnAlertResultListener(new OnAlertResultListener(){
//
//                    @Override
//                    public void onCancel(){
//
//                    }
//
//                    @Override
//                    public void onResult(AlertBuilder.EVENT_ALERT event_alert){
//
//                        switch (event_alert.getValue()) {
//                            case 1:
//                                text_sort.setText(getString(R.string.word_sort_recent));
//                                mSort = EnumData.MobileGiftSortType.recent;
//                                break;
//                            case 2:
//                                text_sort.setText(getString(R.string.word_sort_high_price));
//                                mSort = EnumData.MobileGiftSortType.highPrice;
//                                break;
//                            case 3:
//                                text_sort.setText(getString(R.string.word_sort_low_price));
//                                mSort = EnumData.MobileGiftSortType.lowPrice;
//                                break;
//                        }
//                        getCount();
//                    }
//                }).builder().show(getActivity());
//                break;
//        }
//    }
//
//    private void getCount(){
//
//        Map<String, String> params = new HashMap<>();
//        if(mCategory.getNo() != null){
//            params.put("no", "" + mCategory.getNo());
//        }
//
//        ApiBuilder.create().getMobileGiftCount(params).setCallback(new PplusCallback<NewResultResponse<Integer>>(){
//
//            @Override
//            public void onResponse(Call<NewResultResponse<Integer>> call, NewResultResponse<Integer> response){
//                if(!isAdded()){
//                    return;
//                }
//                mTotalCount = response.getData();
//                text_total_count.setText(getString(R.string.format_total_with_count, mTotalCount));
//                mPage = 1;
//                mAdapter.clear();
//                listCall(mPage);
//            }
//
//            @Override
//            public void onFailure(Call<NewResultResponse<Integer>> call, Throwable t, NewResultResponse<Integer> response){
//
//            }
//        }).build().call();
//    }
//
//    private void listCall(int page){
//
//
//        Map<String, String> params = new HashMap<>();
//        if(mCategory.getNo() != null){
//            params.put("no", "" + mCategory.getNo());
//        }
//        params.put("align", mSort.name());
//        params.put("pg", "" + page);
//        mLockListView = true;
//        showProgress("");
//        ApiBuilder.create().getMobileGiftList(params).setCallback(new PplusCallback<NewResultResponse<MobileGift>>(){
//
//            @Override
//            public void onResponse(Call<NewResultResponse<MobileGift>> call, NewResultResponse<MobileGift> response){
//                if(!isAdded()){
//                    return;
//                }
//                mLockListView = false;
//                hideProgress();
//                mAdapter.addAll(response.getDatas());
//            }
//
//            @Override
//            public void onFailure(Call<NewResultResponse<MobileGift>> call, Throwable t, NewResultResponse<MobileGift> response){
//
//                mLockListView = false;
//                hideProgress();
//            }
//        }).build().call();
//    }
//
//    private class ItemOffsetDecoration extends RecyclerView.ItemDecoration{
//
//        private int mItemOffset, mRightOffset;
//
//        public ItemOffsetDecoration(int itemOffset, int rightOffset){
//
//            mItemOffset = itemOffset;
//            mRightOffset = rightOffset;
//        }
//
//        public ItemOffsetDecoration(@NonNull Context context, @DimenRes int itemOffsetId){
//
//            this(context.getResources().getDimensionPixelSize(itemOffsetId), context.getResources().getDimensionPixelSize(R.dimen.width_8));
//        }
//
//        @Override
//        public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state){
//
//            super.getItemOffsets(outRect, view, parent, state);
//
//            int position = parent.getChildAdapterPosition(view);
//            if(position < 2){
//                outRect.set(mRightOffset, mItemOffset, mRightOffset , mRightOffset);
//            }else{
//                outRect.set(mRightOffset, mRightOffset, mRightOffset , mRightOffset);
//            }
////            if(position == 0){
////
////            }else if(position == 1){
////                outRect.set(mRightOffset, mItemOffset, 0, mRightOffset);
////            }else{
////                if(position % 2 == 1) {
////                    outRect.set(mRightOffset, mRightOffset, 0, mRightOffset);
////                } else {
////                    outRect.set(0, mRightOffset, mRightOffset, mRightOffset);
////                }
////            }
//
//        }
//    }
//}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/mobilegift/ui/MobileGiftListFragment.kt
//package com.pplus.prnumberuser.apps.mobilegift.ui
//
//import android.content.Context
//import android.content.Intent
//import android.graphics.Rect
//import android.os.Bundle
//import android.view.View
//import androidx.annotation.DimenRes
//import androidx.recyclerview.widget.GridLayoutManager
//import androidx.recyclerview.widget.LinearLayoutManager
//import androidx.recyclerview.widget.RecyclerView
//import androidx.recyclerview.widget.RecyclerView.ItemDecoration
//import com.pplus.utils.part.format.FormatUtil
//import com.pplus.prnumberuser.Const
//import com.pplus.prnumberuser.R
//import com.pplus.prnumberuser.apps.common.builder.AlertBuilder
//import com.pplus.prnumberuser.apps.common.builder.AlertBuilder.EVENT_ALERT
//import com.pplus.prnumberuser.apps.common.builder.OnAlertResultListener
//import com.pplus.prnumberuser.apps.common.ui.base.BaseActivity
//import com.pplus.prnumberuser.apps.common.ui.base.BaseFragment
//import com.pplus.prnumberuser.apps.mobilegift.data.MobileGiftAdapter
//import com.pplus.prnumberuser.core.code.common.EnumData.MobileGiftSortType
//import com.pplus.prnumberuser.core.network.ApiBuilder
//import com.pplus.prnumberuser.core.network.model.dto.MobileGift
//import com.pplus.prnumberuser.core.network.model.dto.MobileGiftCategory
//import com.pplus.prnumberuser.core.network.model.response.NewResultResponse
//import com.pplus.prnumberuser.core.util.PplusCommonUtil
//import kotlinx.android.synthetic.main.fragment_mobile_gift_list.*
//import com.pplus.networks.common.PplusCallback
//import retrofit2.Call
//import java.util.*
//
//class MobileGiftListFragment : BaseFragment<BaseActivity?>() {
//    override fun getPID(): String {
//        return ""
//    }
//
//    override fun getLayoutResourceId(): Int {
//        return R.layout.fragment_mobile_gift_list
//    }
//
//    private var mAdapter: MobileGiftAdapter? = null
//    var mLayoutManager: LinearLayoutManager? = null
//    private var mPage = 1
//    private var mTotalCount = 0
//    private var mLockListView = true
//    private var mCategory: MobileGiftCategory? = null
//    private var mSort: MobileGiftSortType? = null
//    private var mPos = 0
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        if (arguments != null) {
//            mCategory = arguments!!.getParcelable(Const.CATEGORY)
//            mPos = arguments!!.getInt(Const.POSITION)
//        }
//    }
//
//
//    override fun initializeView(container: View) {
//
//    }
//
//    override fun init() {
//        mLayoutManager = LinearLayoutManager(activity)
//        recycler_mobile_gift_list.layoutManager = mLayoutManager
////        recycler_mobile_gift_list.addItemDecoration(ItemOffsetDecoration(activity!!, R.dimen.height_16))
//        mAdapter = MobileGiftAdapter()
//        recycler_mobile_gift_list.adapter = mAdapter
//        text_mobile_gift_total_count?.text = getString(R.string.format_count_unit, "0")
////        text_mobile_gift_sort.setOnClickListener {
////            val builder = AlertBuilder.Builder()
////            builder.setContents(getString(R.string.word_sort_recent), getString(R.string.word_sort_high_price), getString(R.string.word_sort_low_price))
////            builder.setStyle_alert(AlertBuilder.STYLE_ALERT.LIST_CENTER).setOnAlertResultListener(object : OnAlertResultListener {
////                override fun onCancel() {}
////                override fun onResult(event_alert: EVENT_ALERT) {
////                    when (event_alert.getValue()) {
////                        1 -> {
////                            text_mobile_gift_sort.text = getString(R.string.word_sort_recent)
////                            mSort = MobileGiftSortType.recent
////                        }
////                        2 -> {
////                            text_mobile_gift_sort.text = getString(R.string.word_sort_high_price)
////                            mSort = MobileGiftSortType.highPrice
////                        }
////                        3 -> {
////                            text_mobile_gift_sort.text = getString(R.string.word_sort_low_price)
////                            mSort = MobileGiftSortType.lowPrice
////                        }
////                    }
////                    getCount()
////                }
////            }).builder().show(activity)
////        }
//        recycler_mobile_gift_list.addOnScrollListener(object : RecyclerView.OnScrollListener() {
//            var pastVisibleItems = 0
//            var visibleItemCount = 0
//            var totalItemCount = 0
//            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
//                super.onScrolled(recyclerView, dx, dy)
//                visibleItemCount = mLayoutManager!!.childCount
//                totalItemCount = mLayoutManager!!.itemCount
//                pastVisibleItems = mLayoutManager!!.findFirstVisibleItemPosition()
//                if (!mLockListView) {
//                    if (totalItemCount < mTotalCount && visibleItemCount + pastVisibleItems >= totalItemCount * Const.NEXT_RECYCLERVIEW_MARGIN) {
//                        mPage++
//                        listCall(mPage)
//                    }
//                }
//            }
//        })
//        mAdapter!!.setOnItemClickListener(object : MobileGiftAdapter.OnItemClickListener{
//            override fun onItemClick(position: Int) {
//                val intent = Intent(activity, MobileGiftDetailActivity::class.java)
//                intent.putExtra(Const.DATA, mAdapter!!.getItem(position))
//                intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP
//                startActivity(intent)
//            }
//        })
//        mSort = MobileGiftSortType.recent
////        text_mobile_gift_sort.setText(R.string.word_sort_low_price)
//
//        getCount()
//    }
//
//    private fun getCount() {
//        val params: MutableMap<String, String> = HashMap()
//        if (mCategory!!.no != null) {
//            params["no"] = "" + mCategory!!.no
//        }
//        ApiBuilder.create().getMobileGiftCount(params).setCallback(object : PplusCallback<NewResultResponse<Int>> {
//            override fun onResponse(call: Call<NewResultResponse<Int>>?, response: NewResultResponse<Int>?) {
//                if (!isAdded) {
//                    return
//                }
//                if (response != null) {
//                    mTotalCount = response.data
//                    text_mobile_gift_total_count?.text = PplusCommonUtil.fromHtml(getString(R.string.html_total_count2, FormatUtil.getMoneyType(mTotalCount.toString())))
//                    mPage = 1
//                    mAdapter!!.clear()
//                    listCall(mPage)
//                }
//
//            }
//
//            override fun onFailure(call: Call<NewResultResponse<Int>>?, t: Throwable?, response: NewResultResponse<Int>?) {
//
//            }
//        }).build().call()
//    }
//
//    private fun listCall(page: Int) {
//        val params: MutableMap<String, String> = HashMap()
//        if (mCategory!!.no != null) {
//            params["no"] = "" + mCategory!!.no
//        }
//        params["align"] = mSort!!.name
//        params["pg"] = "" + page
//        mLockListView = true
////        if (page != 1 || mPos == 0) {
////            showProgress("")
////        }
//        ApiBuilder.create().getMobileGiftList(params).setCallback(object : PplusCallback<NewResultResponse<MobileGift>> {
//            override fun onResponse(call: Call<NewResultResponse<MobileGift>>?, response: NewResultResponse<MobileGift>?) {
////                hideProgress()
//                if (!isAdded) {
//                    return
//                }
//                mLockListView = false
//                if (response != null) {
//                    mAdapter!!.addAll(response.datas)
//                }
//
//            }
//
//            override fun onFailure(call: Call<NewResultResponse<MobileGift>>?, t: Throwable?, response: NewResultResponse<MobileGift>?) {
//                mLockListView = false
////                hideProgress()
//            }
//        }).build().call()
//    }
//
//    private inner class ItemOffsetDecoration(private val mItemOffset: Int, private val mRightOffset: Int) : ItemDecoration() {
//
//        constructor(context: Context, @DimenRes itemOffsetId: Int) : this(context.resources.getDimensionPixelSize(itemOffsetId), context.resources.getDimensionPixelSize(R.dimen.width_8)) {}
//
//        override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
//            super.getItemOffsets(outRect, view, parent, state)
//            val position = parent.getChildAdapterPosition(view)
//            if (position < 2) {
//                outRect[mRightOffset, mItemOffset, mRightOffset] = mRightOffset
//            } else {
//                outRect[mRightOffset, mRightOffset, mRightOffset] = mRightOffset
//            }
//        }
//
//    }
//
//    companion object {
//        fun newInstance(category: MobileGiftCategory?, position: Int): MobileGiftListFragment {
//            val fragment = MobileGiftListFragment()
//            val args = Bundle()
//            args.putParcelable(Const.CATEGORY, category)
//            args.putInt(Const.POSITION, position)
//            fragment.arguments = args
//            return fragment
//        }
//    }
//}